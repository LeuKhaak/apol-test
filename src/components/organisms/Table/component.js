import React from "react";
import * as styles from "./styles.module.scss";
import Row from "src/components/molecules/Row";
import Cell from "src/components/molecules/Cell";
import Loader from "src/components/atoms/Loader";

function Table({ data, deleteUser, editeUser, sortUsers, error, loader }) {
  const headers = [
    ["email", "Email"],
    ["firstName", "First Name"],
    ["lastName", "Last Name"],
    ["position", "Position"],
    ["project", "Project Name"],
    ["longitude", "Longitude"],
    ["latitude", "Latitude"],
  ];
  return (
    <div className={styles.table}>
      <div className={styles.tableTitles}>
        <Cell content="SNO" cellStyle="cellTitle" />

        {headers.map((el) => (
          <div
            key={el[0]}
            style={{ cursor: "pointer" }}
            onClick={() => sortUsers(el[0])}
          >
            <Cell content={el[1]} cellStyle="simbol" />
          </div>
        ))}
        <div className={styles.actionsTitleWrapper}>
          <Cell content="Actions" cellStyle="cellTitleActions" />
        </div>
      </div>
      {error ? (
        <div className={styles.error}>"The request failed!"</div>
      ) : loader ? (
        <Loader loader />
      ) : (
        data &&
        data.map((el, ind) => (
          <Row
            key={ind}
            number={ind + 1}
            data={el}
            deleteUser={deleteUser}
            editeUser={editeUser}
          />
        ))
      )}
    </div>
  );
}

export default Table;
