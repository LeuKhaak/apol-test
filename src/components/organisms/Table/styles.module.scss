@import "../../../assets/styles/styles.scss";
.table {
  .tableTitles {
    @include row;
  }
  .error {
    font-size: 2rem;
    font-weight: 500;
    color: $error;
  }
}
