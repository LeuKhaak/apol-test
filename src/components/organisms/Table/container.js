import { React } from "react";
import Table from "./component";
import { useSelector, useDispatch } from "react-redux";
import { actionUpdateData } from "src/store/actions/usersActions";

function TableContainer() {
  const dispatch = useDispatch();

  const usersList = useSelector((state) => state.users.usersList);
  const error = useSelector((state) => state.users.getError);
  const loader = useSelector((state) => state.users.loader);

  const deleteUser = (id) => {
    // eslint-disable-next-line no-restricted-globals
    const check = confirm("Are you sure?");
    if (!check) return;
    const list = [...usersList];
    const newList = list.filter((item) => item[0][1] !== id);
    dispatch(actionUpdateData(newList));
  };

  const editeUser = (id, row) => {
    const list = [...usersList];
    const ind = list.findIndex((el) => el[0][1] === id);
    list.splice(ind, 1, row);
    dispatch(actionUpdateData(list));
  };

  const sortUsers = (arg) => {
    const list = [...usersList];
    const ind = list[0].findIndex((item) => item[0] === arg);
    list.sort((a, b) => (a[ind] > b[ind] ? 1 : -1));
    dispatch(actionUpdateData(list));
  };

  return (
    <Table
      data={usersList}
      deleteUser={deleteUser}
      editeUser={editeUser}
      sortUsers={sortUsers}
      error={error}
      loader={loader}
    />
  );
}

export const container = TableContainer;
