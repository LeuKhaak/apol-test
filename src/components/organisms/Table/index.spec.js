import React from "react";
import { shallow } from "enzyme";
import Table from "./component";

let component = null;
const setComponent = (props) => shallow(<Table {...props} />);
beforeEach(() => {
  component = setComponent();
});

describe("Table testing:", () => {
  test("is Error shown", () => {
    component = setComponent({
      getError: true,
    });
    expect(component).toMatchSnapshot();
  });

  test("is Loader shown", () => {
    component = setComponent({
      getError: null,
      loader: true,
    });
    expect(component).toMatchSnapshot();
  });

  //test("is CardList rendered", () => {
  //  component = setComponent({
  //    getError: null,
  //    loader: false,
  //    data: [
  //      {
  //        char_id: 1,
  //        name: "Bill",
  //      },
  //      {
  //        char_id: 2,
  //        name: "Cat",
  //      },
  //    ],
  //  });
  //  expect(component).toMatchSnapshot();
  //});
});
