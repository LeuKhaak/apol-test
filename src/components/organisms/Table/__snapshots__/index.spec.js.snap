// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table testing: is Error shown 1`] = `
<div>
  <div>
    <Cell
      cellStyle="cellTitle"
      content="№"
    />
    <div
      key="email"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <Cell
        cellContentStyle="simbol"
        cellStyle="cellTitle"
        content="Email"
      />
    </div>
    <div
      key="firstName"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <Cell
        cellContentStyle="simbol"
        cellStyle="cellTitle"
        content="First Name"
      />
    </div>
    <div
      key="lastName"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <Cell
        cellContentStyle="simbol"
        cellStyle="cellTitle"
        content="Last Name"
      />
    </div>
    <div
      key="position"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <Cell
        cellContentStyle="simbol"
        cellStyle="cellTitle"
        content="Position"
      />
    </div>
    <div
      key="project"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <Cell
        cellContentStyle="simbol"
        cellStyle="cellTitle"
        content="Project Name"
      />
    </div>
    <div
      key="longitude"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <Cell
        cellContentStyle="simbol"
        cellStyle="cellTitle"
        content="Longitude"
      />
    </div>
    <div
      key="latitude"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <Cell
        cellContentStyle="simbol"
        cellStyle="cellTitle"
        content="Latitude"
      />
    </div>
    <div>
      <Cell
        cellStyle="cellTitleActions"
        content="Actions"
      />
    </div>
  </div>
</div>
`;

exports[`Table testing: is Loader shown 1`] = `
<div>
  <div>
    <Cell
      cellStyle="cellTitle"
      content="№"
    />
    <div
      key="email"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <Cell
        cellContentStyle="simbol"
        cellStyle="cellTitle"
        content="Email"
      />
    </div>
    <div
      key="firstName"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <Cell
        cellContentStyle="simbol"
        cellStyle="cellTitle"
        content="First Name"
      />
    </div>
    <div
      key="lastName"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <Cell
        cellContentStyle="simbol"
        cellStyle="cellTitle"
        content="Last Name"
      />
    </div>
    <div
      key="position"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <Cell
        cellContentStyle="simbol"
        cellStyle="cellTitle"
        content="Position"
      />
    </div>
    <div
      key="project"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <Cell
        cellContentStyle="simbol"
        cellStyle="cellTitle"
        content="Project Name"
      />
    </div>
    <div
      key="longitude"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <Cell
        cellContentStyle="simbol"
        cellStyle="cellTitle"
        content="Longitude"
      />
    </div>
    <div
      key="latitude"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <Cell
        cellContentStyle="simbol"
        cellStyle="cellTitle"
        content="Latitude"
      />
    </div>
    <div>
      <Cell
        cellStyle="cellTitleActions"
        content="Actions"
      />
    </div>
  </div>
  <Loader
    loader={true}
  />
</div>
`;
